### PR TITLE
Payments

### PR DESCRIPTION
**Pull request by SamuliSalonen**
tx & wallet implementation 
lets client select properties they need from user models, this way the controllers can focus on executing the operation and client can give proper feedback to user by selecting what information they need.
the api still returns information (such as txid's to indicate if the operation was succesfull or not)

### CODE REVIEW

File: BITCORNService/Utils/DbActions/DbOperations.cs [hunk 1]
```diff
@@ -43,27 +44,25 @@ public static async Task<bool> IsBlockchainTransactionLogged(this BitcornContext
...
             var strategy = dbContext.Database.CreateExecutionStrategy();
             await strategy.ExecuteAsync(async () =>
             {
-                using (var transaction = dbContext.Database.BeginTransaction(IsolationLevel.RepeatableRead))
+                try
                 {
-                    try
-                    {
-                        await dbContext.SaveChangesAsync();
-                        transaction.Commit();
```
Comment (JasonSowers) [inline BITCORNService/Utils/DbActions/DbOperations.cs:58]: should this come before the save?

File: BITCORNService/Models/CornTx.cs [hunk 2]
```diff
@@ -14,8 +14,8 @@ public CornTx()
         public string Platform { get; set; }
         public string TxType { get; set; }
         public decimal? Amount { get; set; }
-        public string SenderId { get; set; }
-        public string ReceiverId { get; set; }
+        public int? SenderId { get; set; }
+        public int ReceiverId { get; set; }
```
Comment (JasonSowers) [inline BITCORNService/Models/CornTx.cs:18]: Why are we changing this?

**Review by JasonSowers — CHANGES_REQUESTED** [review]

**Review by JasonSowers — APPROVED** [review]

File: BITCORNService/Models/CornTx.cs (hunk 2 above)
Comment (SamuliSalonen) [reply BITCORNService/Models/CornTx.cs:18]: corntx should always point to user primary key

File: BITCORNService/Utils/DbActions/DbOperations.cs (hunk 1 above)
Comment (SamuliSalonen) [reply BITCORNService/Utils/DbActions/DbOperations.cs:58]: db transactions always are committed after save

**Pull request merged**